titiler/cmr/timeseries.py

```diff
@@ -0,0 +1,612 @@
+"""Timeseries extension for titiler.cmr"""
+
+import asyncio
+import io
+import re
+from dataclasses import dataclass, fields
+from datetime import datetime, timedelta
+from enum import Enum
+from types import DynamicClassAttribute
+from typing import Annotated, Any, Dict, List, Literal, Optional, Tuple, Union
+from urllib.parse import urlencode
+
+import earthaccess
+import httpx
+from attrs import define
+from dateutil.relativedelta import relativedelta
+from fastapi import Body, Depends, Path, Query, Request, Response
+from fastapi.exceptions import HTTPException
+from fastapi.responses import StreamingResponse
+from geojson_pydantic import Feature, FeatureCollection
+from geojson_pydantic.geometries import Geometry
+from PIL import Image
+from pydantic import BaseModel
+
+from titiler.cmr.dependencies import ConceptID
+from titiler.cmr.factory import Endpoints
+from titiler.core.algorithm import algorithms as available_algorithms
+from titiler.core.dependencies import CoordCRSParams, DefaultDependency, DstCRSParams
+from titiler.core.factory import FactoryExtension
+from titiler.core.models.mapbox import TileJSON
+from titiler.core.models.responses import Statistics
+from titiler.core.resources.enums import ImageType
+from titiler.core.resources.responses import GeoJSONResponse
+
+timeseries_img_endpoint_params: Dict[str, Any] = {
+    "responses": {
+        200: {
+            "content": {
+                "image/gif": {},
+            },
+            "description": "Return an image.",
+        }
+    },
+    "response_class": Response,
+}
+
+
+class TimeseriesMediaType(str, Enum):
+    """Responses Media types formerly known as MIME types."""
+
+    gif = "image/gif"
+
+
+class TimeseriesImageType(str, Enum):
+    """Available output image type."""
+
+    gif = "gif"
+
+    @DynamicClassAttribute
+    def mediatype(self):
+        """Return image media type."""
+        return TimeseriesMediaType[self._name_].value
```

Comment:
TODO: remove once we update titiler >=0.19 https://github.com/developmentseed/titiler/pull/1018